### ISSUE TITLE
Crash on startup - fails to connect to running Outlook instance

### ISSUE THREAD

**Issue by phw198** [phw198/OutlookGoogleCalendarSync#752]
**OGCS Version**: v2.8.0.0  
**Installed or Portable**: Installed

**Problem Description**  
On starting OGCS, it crashes straight away when Outlook is already running.

**Steps to Reproduce the Issue**  
1. Install OGCS
1. Crashes with `System.NullReferenceException`

**Any other information**  
[OGcalsync.log](https://github.com/phw198/OutlookGoogleCalendarSync/files/3266182/OGcalsync.-.Shari.Ruckh.log)
```
2019-06-07 06:32:46,542 INFO  1 OutlookGoogleCalendarSync.ErrorFlagAppender [0] -  Attaching to the already running Outlook process.
2019-06-07 06:32:46,542 WARN  1 OutlookGoogleCalendarSync.ErrorFlagAppender [0] -  Early binding to Outlook appears to have failed.
2019-06-07 06:32:46,543 ERROR  1 OutlookGoogleCalendarSync.ErrorFlagAppender [0] -  System.NullReferenceException: Object reference not set to an instance of an object.
2019-06-07 06:32:46,809 ERROR  1 OutlookGoogleCalendarSync.OGCSexception [0] -  Code: 0x80004003;-2147467261
2019-06-07 06:32:46,816 ERROR  1 OutlookGoogleCalendarSync.OGCSexception [0] -     at OutlookGoogleCalendarSync.OutlookOgcs.Calendar.PoorlyOfficeInstall(Exception caughtException)
2019-06-07 06:32:46,818 DEBUG  1 OutlookGoogleCalendarSync.OutlookOgcs.Calendar [0] -  Could try late binding??
2019-06-07 06:32:46,819 FAIL  1 OutlookGoogleCalendarSync.Telemetry [0] -  Failed determining Outlook client version.
```

**Comment by phw198** [comment]
@meme10506 It looks like when OGCS tries to attach to the already running instance of Outlook, it is unable. Is there anything special about the way Outlook is running, or how it was started?

Does OGCS start properly if you first close down Outlook?

**Comment by meme10506** [comment]
Nothing special about it.  I just double click on the Outlook icon and it opens and runs as usual.  I just tried closing outlook and trying, and got the exact same error message.

**Comment by meme10506** [comment]
Oh, wait, here's a new message...see attached
![Capture](https://user-images.githubusercontent.com/51479169/59117857-cba21f00-891c-11e9-95b5-227db6cdb1ea.JPG)

**Comment by meme10506** [comment]
What is the wiki?

**Comment by phw198** [comment]
Once you click OK, it should automatically open the wiki in your browser with a solution for that error?

**Comment by meme10506** [comment]
Ah, sorry.  You have to be patient with me.  

So, I did a repair on my Office install, reran it, and get the same error.  

I'm so frustrated because I had an app that worked really well, but then my android did a system update and the app is no longer able to work with my phone.  I hope I can get this to work somehow.

**Comment by meme10506** [comment]
Oh, fyi, this is a personal computer, not in a corporate environment.  No IT department to help me.

**Comment by meme10506** [comment]
And I'm using Outlook 2013

**Comment by phw198** [comment]
Could you provide your logfile again with the newer error (when OGCS is started without Outlook running)?

Did the Office Repair prompt you to reboot? If not, try a reboot too...

**Comment by meme10506** [comment]
[OGcalsync.log](https://github.com/phw198/OutlookGoogleCalendarSync/files/3266922/OGcalsync.log)

**Comment by meme10506** [comment]
It did not say to reboot, but I'll give it a try.

**Comment by meme10506** [comment]
Nope.  Didn't help.

**Comment by phw198** [comment]
Thanks - it seems you are actually encountering a different error (not the `0x80004002` which is being reported back). I will need to investigate why this is happening...

**Comment by phw198** [comment]
:package: Please could you apply [hotfix v2.8.02.zip](https://github.com/phw198/OutlookGoogleCalendarSync/files/3267438/v2.8.02.zip) ([see instructions for applying](https://github.com/phw198/OutlookGoogleCalendarSync/wiki/Reporting-Problems#applying-a-hotfix)) and then try:

1. Running OGCS when Outlook is already open.
1. Running OGCS with Outlook shut down.

Hopefully the logfix will record some extra information for me, and may also return a different error code too. Please upload your `OGcalsync.log` file again.

Thanks for your persistence here - really appreciate it! Most people just give up and so I never get the chance to fix this type of problem...

**Comment by meme10506** [comment]
My antivirus doesn't want me to run the program...

**Comment by phw198** [comment]
Is there an option with the anti virus program to let it through? If you can't see how to do that I could "code sign" the hotfix (like I do for the main releases) to see if that makes the anti virus happier, but it's a bit of a process I'd prefer to not go through if possible...

Basically, AV can block OGCS because it automatically connects to Outlook and accesses meeting attendee email addresses etc so it thinks it's "bad" 😒

**Comment by phw198** [comment]
:package: I've gone ahead and code-signed [hotfix v2.8.02.zip](https://github.com/phw198/OutlookGoogleCalendarSync/files/3268437/v2.8.02.zip) ([see instructions for applying](https://github.com/phw198/OutlookGoogleCalendarSync/wiki/Reporting-Problems#applying-a-hotfix)) - please see if that satisfies your anti-virus.

If not though, you'll need to work out how to make your AV allow it, or temporarily disable the AV.

**Comment by phw198** [comment]
Hi @meme10506 - were you able to get the hotfix running?

**Comment by PhabGuy** [comment]
Greetings! I just found OGCS today and I'm experiencing the same issue as @meme10506 above. I'm attempting to install OGCS v2.8.0-beta. I have run an Office Quick Repair, no success. Then an Office Online Repair, also no success. Even after re-boot, no success. I am attaching the OGCS log.

[OGcalsync.log](https://github.com/phw198/OutlookGoogleCalendarSync/files/3390754/OGcalsync.log)

**Comment by phw198** [comment]
@PhabGuy As with @meme10506, you'd need to apply the hotfix [as posted above](https://github.com/phw198/OutlookGoogleCalendarSync/issues/752#issuecomment-500114996), and see if it reports a different error code and associated fix in the wiki.

**Comment by phw198** [comment]
[Hotfix is available](https://github.com/phw198/OutlookGoogleCalendarSync/issues/783#issuecomment-515713570) on #783

**Comment by PhabGuy** [comment]
Thanks Paul. Thanks for your support. I applied the Hotfix. This identified an error that took me to the Wiki for this issue:
 
0x80029C4A - TYPE__E__CANTLOADLIBRARY
 
I starting working through the registry checks and found one Win32 entry that had to be deleted and I had to register MSOUTL.OLB. After doing this, OGCS started functioning.
 
I have one Google Calendar I want to sync with my Outlook, but I do not want to sync all my default Outlook calendar entries to that Google Calendar. Therefore, I created a separate Outlook calendar solely for the purpose of syncing to that Google Calendar. Once this was set up, I started running some tests to gain confidence in the solution.
 
I am now encountering the following error:
Update event failed to save.
Start and end times must either both be date or both be dateTime.
[400=BadRequest]
 
So now I’m investigating that issue.
 
Cheers!
Brad
